dask/dataframe/core.py

```diff
@@ -4141,6 +4141,8 @@ def join(
         npartitions=None,
         shuffle=None,
     ):
+        if is_series_like(other) and hasattr(other, "name"):
+            other = other.to_frame()
```

Comment:
It'd be nice to check for indexes here as well and convert those to dataframes.

Reply:
On `Pandas` it will cause an error. (join indexes only using concat)
```python
>>> df = pd.DataFrame({"x": [1, 2, 3, 4, 5, 6, 7, 8]})
>>> df.join(df.index)

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-31-5499579ebbb6> in <module>
----> 1 df.join(df.index)

~\anaconda3\lib\site-packages\pandas\core\frame.py in join(self, other, on, how, lsuffix, rsuffix, sort)
   7206         5  K5  A5  NaN
   7207         """
-> 7208         return self._join_compat(
   7209             other, on=on, how=how, lsuffix=lsuffix, rsuffix=rsuffix, sort=sort
   7210         )

~\anaconda3\lib\site-packages\pandas\core\frame.py in _join_compat(self, other, on, how, lsuffix, rsuffix, sort)
   7240             frames = [self] + list(other)
   7241 
-> 7242             can_concat = all(df.index.is_unique for df in frames)
   7243 
   7244             # join indexes only using concat

~\anaconda3\lib\site-packages\pandas\core\frame.py in <genexpr>(.0)
   7240             frames = [self] + list(other)
   7241 
-> 7242             can_concat = all(df.index.is_unique for df in frames)
   7243 
   7244             # join indexes only using concat

AttributeError: 'int' object has no attribute 'index'
```

Reply:
Ah ok! I guess this is good to go then :)